src/PimApi/PimCatalogApiClient.php

```diff
@@ -37,16 +38,39 @@ private function getClient(): HttpClientInterface
         return $this->client;
     }
 
+    private function getPimUrl(): string
+    {
+        $pimUrl = $this->pimURLStorage->getPimURL();
+        if (null === $pimUrl) {
+            throw new \LogicException('Can\'t retrieve PIM url, please restart the authorization process.');
+        }
+
+        return $pimUrl;
+    }
+
+    /**
+     * @throws PimApiUnauthorizedException
+     * @throws PimApiException
+     */
+    private function throwOnErroneousResponse(int $expectedCode, int $actualCode, string $message): void
```

Comment:
```suggestion
    private function throwCustomException(int $expectedCode, int $actualCode, string $message): void
```
`throwOnErroneousResponse` is hard to read/understand IMHO.

Reply:
Would `throwOnErrorCode` be ok ?